framework/src/main/java/org/apache/drill/test/framework/Utils.java

```diff
@@ -230,6 +270,27 @@
 	    }
 	    return list;
 	  }
+
+  private static List<File> searchFolders(File root, String regex) {
+          List<File> list = new ArrayList<File>();
+          Pattern pattern = Pattern.compile(regex + "$");
+          Matcher matcher = null;
+          if (!root.isFile()) {
```

Comment:
"root" or "pathPrefix" will always be a directory - as we create it in the very beginning, if I'm not mistaken. Did you intend to do that check for "regex" or "pathList"?